libimage/layer_tree.go

```diff
@@ -22,6 +23,10 @@ type layerTree struct {
 	// emptyImages do not have any top-layer so we cannot create a
 	// *layerNode for them.
 	emptyImages []*Image
+	// manifestList keep track of images based on their digest.
+	// Library will use this map when checking if a image is dangling.
+	// If an image is used in a manifestList it is NOT dangling
+	manifestListDigests map[digest.Digest]bool
```

Comment:
this just be `map[digest.Digest]struct{}`, @mtrmac comment this above already.
The bool needs an extra allocation AFAIK. With struct{} it is clear that is is a set and the value is ignored. With bool you might can set it to true or false so the meaning is not as obvious.

c/image has https://github.com/containers/image/blob/main/internal/set/set.go to abstract a set type, maybe we should make this public for more users if we are concerned about usability

Reply:
I’ve been betting that the standard library will add a Set type Any Time Now, and that adding non-standard set implementations would add risk that they will appear in the public API and we will have to maintain the set implementation even as the standard library adds a different one.

That bet is looking increasingly dubious over time… maybe something will finally happen in https://github.com/golang/go/issues/69230 .

*shrug* I won’t initiate this, but if someone wanted to add a set type to c/storage, I’d migrate c/image to it.